File: iter2/src/System/SystemDomain.java
package System;
import Creator.CreateAdvisor;
import Creator.CreateCourse;
import Creator.CreateLecturer;
import Creator.CreatePage;
import Creator.CreateStudent;

import org.json.JSONException;
import java.io.IOException;


public class SystemDomain {
    private CreateLecturer lecturerCreator;
    private CreateAdvisor advisorCreator;
    private CreateCourse courseCreator;
    private CreateStudent studentCreator;
    private CreatePage pageCreator;

    public SystemDomain() throws JSONException, IOException {
        lecturerCreator = new CreateLecturer("src\\JSON_Files\\lecturers.json");
        advisorCreator = new CreateAdvisor("src\\JSON_Files\\advisors.json",lecturerCreator.getLecturers());
        courseCreator = new CreateCourse("src\\JSON_Files\\courses.json",lecturerCreator.getLecturers());
        studentCreator = new CreateStudent("src\\JSON_Files\\Students\\","src\\JSON_Files\\student_json.txt",courseCreator.getCourses(),advisorCreator.getAdvisors());
        pageCreator = new CreatePage();
    }


    public CreateLecturer getLecturerCreator() {return lecturerCreator;}

    public CreateAdvisor getAdvisorCreator() {return advisorCreator;}

    public CreateCourse getCourseCreator() {return courseCreator;}

    public CreateStudent getStudentCreator() {return studentCreator;}

	public CreatePage getPageCreator() {return pageCreator;}
}

File: python_development/System/SystemDomain.py
from python_development.Creator.CreateAdvisor import CreateAdvisor
from python_development.Creator.CreateCourse import CreateCourse
from python_development.Creator.CreateLecturer import CreateLecturer
from python_development.Creator.CreateStudent import CreateStudent


class CreatePage:
    pass


class SystemDomain:
    def __init__(self):
        self.lecturerCreator = CreateLecturer("JSON_Files\\lecturers.json")
        self.advisorCreator = CreateAdvisor("JSON_Files\\advisors.json", self.lecturerCreator.getLecturers())
        self.courseCreator = CreateCourse("src/JSON_Files/courses.json", self.lecturerCreator.getLecturers())
        self.studentCreator = CreateStudent("JSON_Files\\Students\\", "JSON_Files\\student_json.txt",
                                             self.courseCreator.getCourses(), self.advisorCreator.getAdvisors())
        self.pageCreator = CreatePage()

    def getLecturerCreator(self):
        return self.lecturerCreator

    def getAdvisorCreator(self):
        return self.advisorCreator

    def getCourseCreator(self):
        return self.courseCreator

    def getStudentCreator(self):
        return self.studentCreator

    def getPageCreator(self):
        return self.pageCreator
